mountpoint-s3/src/inode/readdir.rs

```diff
@@ -491,6 +516,10 @@ mod ordered {
         fn next(&mut self) -> Option<ReaddirEntry> {
             self.entries.pop_front()
         }
+
+        fn rewind(&mut self) {
+            self.entries.clear();
+        }
```

Comment:
Not sure this is right — we need to include the local entries again if we're going back to 0, but this throws them all away. This kind of subtlety is why I think it might be better to just throw the whole handle away and start again.

Reply:
Either way, let's make sure we have a test covering this case. Something like a directory whose contents are just one local file, iterate the handle once and then rewind it.

Reply:
There is a test that adds a local and a remote file after `opendir` and those are reflected in `readdir` after rewinding. I'll add one additional test that covers just the local case as you mentioned.

Regardless, let me try to just recreate everything which should simplify things.